classes/requests/helpers/class-kco-request-options.php

```diff
@@ -89,28 +89,24 @@ private function get_dob_mandatory() {
 	 * @return string
 	 */
 	private function get_allowed_customer_types() {
-		// Allow external payment method plugin to do its thing.
-		// @TODO: Extract this into a hooked function.
-		if ( in_array( $this->get_purchase_country(), array( 'SE', 'NO', 'FI' ), true ) ) {
-			if ( isset( $this->settings['allowed_customer_types'] ) ) {
-				$customer_types_setting = $this->settings['allowed_customer_types'];
-
-				switch ( $customer_types_setting ) {
-					case 'B2B':
-						$allowed_customer_types = array( 'organization' );
-						break;
-					case 'B2BC':
-						$allowed_customer_types = array( 'person', 'organization' );
-						break;
-					case 'B2CB':
-						$allowed_customer_types = array( 'person', 'organization' );
-						break;
-					default:
-						$allowed_customer_types = array( 'person' );
-				}
 
-				return $allowed_customer_types;
+		if ( null !== in_array( $this->get_purchase_country(), array( 'SE', 'NO', 'FI' ), true ) && isset( $this->settings['allowed_customer_types'] ) ) {
```

Comment:
in_array returns a boolean, checking if its not null will cause it to never fail on the in_array part of this check. Also to check a boolean you do not need the equal marks. Simply do if( in_array() && isset() ) { ... }